text/000-buildpacks-contribute-default-process-type.md

```diff
@@ -0,0 +1,138 @@
+
+Example: multi-process app with an earlier buildpack declaring a process type of `web` with `default = true` and a later buildpack redefining what `web` means WITH `default = true` -> the app image will have a default process type of `web`, with the later buildpack's definition of web.
+
+- Define the target persona: buildpack author, buildpack user.
+
+New buildpack users would need to be taught that buildpacks will be responsible for setting a sensible default process type. They should know how to override the buildpacks-provided default if they desire.
+
+Existing buildpack users would need to be taught about the change so that (1) they no longer have to pass a flag to specify the default process type or (2) they now need to pass a flag to explicitly override the default that buildpacks choose. For (2), it's likely that the number of users in this category would be small given that the lifecycle (pre- platform API 0.4) and pack (post- platform API 0.4) both encourage `web` as the default, and it's likely that buildpacks would also encourage this same default.
+
+Buildpack authors would need to be taught how to configure the default process type.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The changes from this RFC should require a bump to the buildpack API.
+
+Currently, during the `build` phase, buildpacks may declare process types by writing entries to `<layers>/launch.toml`. During the `export` phase, the `<layers>/launch.toml` from each buildpack are aggregated to produce a combined processes list "such that process types from later buildpacks override identical process types from earlier buildpacks" ([spec](https://github.com/buildpacks/spec/blob/main/buildpack.md#process-3)).
+
+This RFC proposes a new `default` key in `<layers>/launch.toml`:
+
+```
+[[processes]]
+type = "<process type>"
+command = "<command to run>"
+default = <boolean (default=false)>
+```
+
+Example:
+
+```toml
+[[processes]]
+type = "web"
+command = "bundle exec ruby app.rb"
+default = true
+
+[[processes]]
+type = "worker"
+command = "bundle exec ruby worker.rb"
+```
+
+Buildpacks may set `default = true` to indicate that the process type being defined should be the default process type for the app image. If not specified, no default designation will be assumed (to be consistent with buildpacks implementing earlier buildpack APIs).
+
+If a buildpack attempts to define two processes with `default = true` specified, the lifecycle will fail.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+- Buildpacks could stomp on each other, each trying to declare different default process types. This might be confusing.
+- Changing the way things work now could surprise some people.
+- One more thing for buildpack authors to know and care about.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- What other designs have been considered?
+  - Keep things as they are.
+  - Earlier iterations of this PR.
```

Comment:
could we capture a brief summary of some of those iterations? in particular override

Reply:
Added here: https://github.com/buildpacks/rfcs/pull/110/files#diff-00a2fb620563b974f876628cc19e746a4a96606f9dea0fd58dc1bb13f010b4abR113-R120